spec/index.html

```diff
@@ -259,28 +255,100 @@ <h2>A Canonical form of N-Quads</h2>
       and <code>graphLabel</code> if present, MUST be a single space, 
       (<code>U+0020</code>).  All other locations that allow 
       white space MUST be empty.</li>
-    <li>There MUST be no comments.</li>
-    <li><code>HEX</code> MUST use only uppercase letters (<code>[A-F]</code>).</li>
-    <li>Characters MUST NOT be represented by <code>UCHAR</code>.</li>
+    <li><a data-cite="RDF12-CONCEPTS#literal">Literals</a> with the
+      datatype <code>http://www.w3.org/2001/XMLSchema#string</code>
+      MUST NOT use the datatype IRI part of the <a href="#grammar-production-literal">literal</a>,
+      and are represented using only <a href="#grammar-production-STRING_LITERAL_QUOTE">STRING_LITERAL_QUOTE</a>.
+    </li>
+    <li>There MUST be no comments or <a>blank lines</a>.</li>
+    <!--li><code><a href="#grammar-production-HEX">HEX</a></code> MUST use only uppercase letters (<code>[A-F]</code>).</li-->
+    <li>Characters MUST NOT be represented by <code><a href="#grammar-production-UCHAR">UCHAR</a></code>.</li>
```

Comment:
Needs discussion.

This is problematic in two ways:

Unfortunately, the IRI rule allows more than the characters of a legal IRI so if the goal is a canonical form that corresponds to the grammar (I think it should), it has to cope with this situation.

Raw backspace, tabs, null and form-feed anywhere in the canonical form leads to obfuscation attacks. If this canonical form is used for RCH, these will need to be addresses or a positive justification made for them. Otherwise, the security considerations sections will be advising differently to the canonical representation.

It would be good if there were a simple check to see whether a claimed canonical form actually hiding an obfuscation. One way is the presences of any the most dubious codepoints.

More for #16.

Reply:
I'll leave in place for now and address as part of #16, which is more comprehensive. I'll add text to the issue marker in the spec.

Reply:
Check updated first point.